NangoHQ/nango
# feat(logs): timeout old operations

## Describe your changes

Fixes NAN-1036

Since we have only 15days retention and syncs are unlimited it's mostly for cosmetic and some smaller better time scoped operations (action, oauth, webhook).

- Add expiration date to all operations
- Cron to set `timeout` to all expired operations

## How to test?

- Trigger an oauth but don't complete the process and wait 5-10minutes
or
- Add an operation, change the expiresAt manually, wait for the cron

(The query in curl)
```sh
#!/usr/bin/env bash

curl -X POST "http://localhost:55132/20240528_messages.2024-05-30/_search" -H 'Content-Type: application/json' -d'
{
  "query": {
            "bool": {
                "must": [{ "range": { "expiresAt": { "lt": "now" } } }],
                "must_not": { "exists": { "field": "parentId" } },
                "should": [{ "term": { "state": "waiting" } }, { "term": { "state": "running" } }]
            }
        }
}
'
```
